src/gr-ingredients-list.c

```diff
@@ -279,16 +279,14 @@ gr_ingredients_list_get_unit (GrIngredientsList *ingredients,
 
 double
 gr_ingredients_list_get_amount (GrIngredientsList *ingredients,
-                                const char        *segment,
                                 const char        *name)
 {
         GList *l;
 
         for (l = ingredients->ingredients; l; l = l->next) {
                 Ingredient *ing = (Ingredient *)l->data;
 
-                if (g_strcmp0 (segment, ing->segment) == 0 &&
-                    g_strcmp0 (name, ing->name) == 0) {
+                if (g_strcmp0 (name, ing->name) == 0) {
```

Comment:
This change looks unrelated, and ... wrong. Why do we stop comparing the segment here ?